src/inspect_ai/tool/_tools/_web_browser/_web_browser.py

```diff
@@ -382,6 +397,58 @@ async def web_browser_cmd(cmd: str, *args: str) -> str:
                 f"web_browser output must contain either 'error' or 'web_at' field: {result.stdout}"
             )
 
+async def web_browser_cmd_local(cmd: str, *args: str) -> str:
+    session_flag = ""
+    browser_session = store().get(BROWSER_SESSION_ID, "")
+    
+    if not browser_session:
+        if not os.path.exists(LOCAL_WEB_CLIENT_NEW_SESSION):
+            raise RuntimeError("Local web browser new session script not found")
+            
+        process = await asyncio.create_subprocess_exec(
+            sys.executable,
+            LOCAL_WEB_CLIENT_NEW_SESSION,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE
+        )
+        stdout, stderr = await process.communicate()
+        
+        if process.returncode != 0:
+            raise RuntimeError(f"Error creating new web browser session: {stderr.decode()}")
+            
+        browser_session = stdout.decode().strip("\n")
+        store().set(BROWSER_SESSION_ID, browser_session)
+    
+    session_flag = f"--session_name={browser_session}"
+    
+    process = await asyncio.create_subprocess_exec(
+        sys.executable,
+        LOCAL_WEB_CLIENT_REQUEST,
+        session_flag,
+        cmd,
+        *args,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE
+    )
+    stdout, stderr = await process.communicate()
```

Comment:
curious if these can be called natively by importing their methods rather than via subprocess -- happy to keep as is if not

Reply:
I'm trying something along the lines of:

```
from inspect_ai.tool._tools._web_browser._resources import web_server  # type: ignore
...
web_server_thread = threading.Thread(target=web_server.main, daemon=True)
    web_server_thread.start()
```

but I run into a bunch of module import errors when trying to import web_server in this way; stuff in from inspect_ai.tool._tools._web_browser._resources that I need to import 🤔

Reply:
I think we need to call in a thread because otherwise the web server execution blocks the main thread

Reply:
Ah whoops this is on the wrong thread

Reply:
Flagging with TODO [add TODO](https://github.com/james-aung/inspect_ai/pull/1/commits/d3c77235bfea64d8534575e6915c1b452203dcba)